### ISSUE TITLE
[BUG] mpegts no PTS found at end of file, duration not set

### ISSUE THREAD

**Issue by SijmenHuizenga** [livekit/egress#530]
**Describe the bug**
_Some_ (not all) `.ts` files created by egress have no [Presentation timestamp (PTS)](https://en.wikipedia.org/wiki/Presentation_timestamp) set.

**Egress Version**
We are use Livekit Cloud. This bug was observed with a recording of 2023-11-07 15:44:21:624 UTC.

**Behavior**
I'm attaching 2 `.ts` files from the same egress recording.

* testbed_20231107154421624.ts -> `no PTS found at end of file, duration not set`
* testbed_20231107154427624.ts -> no errors, everything in order

[pts-missing.zip](https://github.com/livekit/egress/files/13294346/pts-missing.zip)

![image](https://github.com/livekit/egress/assets/8008227/0a230b00-55f0-4059-bae0-30bd70c13cc7)

I would expect all files to be timestamped in the same way. If this is expected, could you maybe explain why we're seeing this?

**Additional context**

This recording was made in cloud project `p_2u70izoj75h` egresses `EG_hzVc5XuBuwp7`.

**Comment by frostbyte73** [comment]
Looks like there was some packet loss surrounding that segment. Does this break the playlist?

**Comment by SijmenHuizenga** [comment]
Yes, this breaks playback using [video.js](https://github.com/videojs/video.js) and [hls.js](https://github.com/video-dev/hls.js). The video gets stuck into an endless buffer state.

**Comment by SijmenHuizenga** [comment]
After some more testing it seems that this specific error does not stop video.js from replaying the video.
